paddle/fluid/framework/ir/mkldnn/requant_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,353 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/mkldnn/requant_mkldnn_fuse_pass.h"
+#include <float.h>
+#include <algorithm>
+#include "paddle/fluid/framework/ir/graph_helper.h"
+#include "paddle/fluid/framework/op_version_registry.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+void RequantMkldnnFusePass::GetTensorFromVector(
+    const std::vector<float>& data_v, Tensor* tensor) const {
+  tensor->mutable_data<float>({static_cast<int>(data_v.size())},
+                              platform::CPUPlace());
+  auto dev_ctx = paddle::platform::CPUDeviceContext();
+  TensorFromVector(data_v, dev_ctx, tensor);
+}
+
+void RequantMkldnnFusePass::GetQuantInfo(
+    ir::Graph* graph, Scope* scope, StringTensorPtrMap& weight_thresholds,
+    StringPairMap& var_quant_scales) const {
+  for (auto* op_node :
+       ir::TopologyVarientSort(*graph, static_cast<ir::SortKind>(0))) {
+    if (!op_node->IsOp()) continue;
+
+    auto* op_desc = op_node->Op();
+    if (op_desc->GetAttrIfExists<bool>("has_quant_info")) {
+      op_desc->RemoveAttr("has_quant_info");
+      std::vector<std::string> attr_names = op_desc->AttrNames();
+      for (auto fake_name : attr_names) {
+        if (fake_name.find("_weight_thresholds") != std::string::npos) {
+          size_t pos = fake_name.find("_weight_thresholds");
+          std::string name = fake_name.substr(0, pos);
+          auto thresholds_vector =
+              BOOST_GET_CONST(std::vector<float>, op_desc->GetAttr(fake_name));
+
+          Tensor tensor;
+          GetTensorFromVector(thresholds_vector, &tensor);
+          weight_thresholds[name] = &tensor;
+          op_desc->RemoveAttr(fake_name);
+        }
+
+        if (fake_name.find("_var_quant_scales") != std::string::npos) {
+          size_t pos = fake_name.find("_var_quant_scales");
+          std::string name = fake_name.substr(0, pos);
+          auto scales_vector =
+              BOOST_GET_CONST(std::vector<float>, op_desc->GetAttr(fake_name));
+
+          Tensor tensor;
+          GetTensorFromVector(scales_vector, &tensor);
+          var_quant_scales[name] = std::make_pair(false, &tensor);
+          op_desc->RemoveAttr(fake_name);
+        }
+      }
+      break;
+    }
+  }
+}
+
+void RequantMkldnnFusePass::ComputeWeightScales(
+    ir::Graph* graph, Scope* scope, StringPairMap& var_quant_scales) const {
+  auto get_scales = [&](Tensor* tensor, int axis) -> std::vector<float> {
```

Comment:
Could you change all lambdas to function?  It will simplify the testing process.

Reply:
OK, I will do it as soon as possible.